### DISCUSSION TITLE
How to extend 'handle' property without issues

### DISCUSSION THREAD

**Discussion by winterdominik** [remix-run/remix#5460] [Q&A]
We are trying to extend the interface (handle property) found in the docu an run into the following issue:

Has some one encounter a similiar problem or found a workaround?

```typescript
interface RouteObject {
  path?: string;
  index?: boolean;
  children?: React.ReactNode;
  caseSensitive?: boolean;
  id?: string;
  loader?: LoaderFunction;
  action?: ActionFunction;
  element?: ReactNode | null;
  errorElement?: ReactNode | null;
  handle?: { lable: string };
  shouldRevalidate?: ShouldRevalidateFunction;
}
```

```
    Type 'RouteObject' is not assignable to type 'NonIndexRouteObject'.
      Types of property 'index' are incompatible.
        Type 'boolean' is not assignable to type 'false'.
```

**Comment by paul-vd** [comment]
I'm also looking for a solution, it seems like it's not possible because  `type RouteHandle` is not exported by `@remix-run/server-runtime`.

If it were, then we could do something simple like

```ts
import "@remix-run/server-runtime";

declare module "@remix-run/server-runtime" {
  type RouteHandle = {
     [key:string]: any,
     label?: string
   };
}
```

based off of the example from https://sergiodxa.com/tutorials/customize-remix-app-load-context-type

for reference:
- [index.ts](https://github.com/remix-run/remix/blob/main/packages/remix-server-runtime/index.ts)
- [reexport.ts](https://github.com/remix-run/remix/blob/main/packages/remix-server-runtime/reexport.ts)
- [routeModules.ts](https://github.com/remix-run/remix/blob/main/packages/remix-server-runtime/routeModules.ts#L255)

**Reply by kiliman** [reply]
The `handle` export is basically of type `any` or `unknown`. Remix knows nothing about it other than providing you with a way to store something per route.

How are you using it so that you can't just cast it to the correct type?
